lib/msal-common/test/unifiedCache/mockStorage.ts

```diff
@@ -0,0 +1,286 @@
+                cache.accounts[key] = value as AccountEntity;
+                break;
+            }
+            case CacheSchemaType.CREDENTIAL: {
+                const credentialType = CacheHelper.getCredentialType(key);
+                switch (credentialType) {
+                    case CredentialType.ID_TOKEN: {
+                        cache.idTokens[key] = value as IdTokenEntity;
+                        break;
+                    }
+                    case CredentialType.ACCESS_TOKEN: {
+                        cache.accessTokens[key] = value as AccessTokenEntity;
+                        break;
+                    }
+                    case CredentialType.REFRESH_TOKEN: {
+                        cache.refreshTokens[key] = value as RefreshTokenEntity;
+                        break;
+                    }
+                }
+                break;
+            }
+            case CacheSchemaType.APP_META_DATA: {
+                cache.appMetadata[key] = value as AppMetadataEntity;
+                break;
+            }
+            default: {
+                console.log("Invalid Cache Type");
+                return;
+            }
+        }
+
+        // update inMemoryCache
+        this.setCache(cache);
+    }
+
+    /**
+     * Gets cache item with given key.
+     * Will retrieve frm cookies if storeAuthStateInCookie is set to true.
+     * @param key
+     * @param type
+     * @param inMemory
+     */
+    getItem(key: string, type?: string, inMemory?: boolean): string | object {
+        // check memory type
+        if (!inMemory) {
+            console.log("Node doesn't support granular cache persistence yet");
+            return {};
+        }
+
+        // read inMemoryCache
+        const cache = this.getCache();
+
+        // save the cacheItem
+        switch (type!) {
+            case CacheSchemaType.ACCOUNT: {
+                return (cache.accounts[key] as AccountEntity) || null;
+            }
+            case CacheSchemaType.CREDENTIAL: {
+                const credentialType = CacheHelper.getCredentialType(key);
+                let credential = null;
+                switch (credentialType) {
+                    case CredentialType.ID_TOKEN: {
+                        credential =
+                            (cache.idTokens[key] as IdTokenEntity) || null;
+                        break;
+                    }
+                    case CredentialType.ACCESS_TOKEN: {
+                        credential =
+                            (cache.accessTokens[key] as AccessTokenEntity) ||
+                            null;
+                        break;
+                    }
+                    case CredentialType.REFRESH_TOKEN: {
+                        credential =
+                            (cache.refreshTokens[key] as RefreshTokenEntity) ||
+                            null;
+                        break;
+                    }
+                }
+                return credential!;
```

Comment:
Instead of breaking and mutating `credential` in each `case`, you can just return in each case.

Reply:
I will fix this in a later PR.